transcoder/output/json/JsonOutputManager.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright 2022 Google LLC
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import os
+import sys
+
+#from genson import SchemaBuilder
+from transcoder.message import DatacastSchema, DatacastField
+from transcoder.output import OutputManager
+
+class JsonOutputManager(OutputManager):
+    """Transcode messages to JSON encoding persisted to a file"""
+    def __init__(self, prefix: str, output_path: str, lazy_create_resources: bool = False):
+        super().__init__(lazy_create_resources=lazy_create_resources)
+        self.prefix = prefix
+        self.schemas = {}
+        self.writers = {}
+
+        if output_path is None:
+            rel_path = "jsonOut"
+            main_script_dir = os.path.dirname(sys.argv[0])
+            self.output_path = os.path.join(main_script_dir, rel_path)
+        else:
+            self.output_path = output_path
+
+        exists = os.path.exists(self.output_path)
+        if not exists:
+            os.makedirs(self.output_path)
+
+    @staticmethod
+    def init_schema():
+        schema = {}
+        schema['$schema'] = 'https://json-schema.org/draft/2019-09/schema'
+        schema['type'] = 'object'
+        return schema
+
+    @staticmethod
+    def output_type_identifier():
+        return 'json'
+
+    def _create_field(self, field: DatacastField):
+        return field.create_json_field()
+
+    def _add_schema(self, schema: DatacastSchema):
+
+        if schema.name in self.schemas:
+            del self.schemas[schema.name]
+        if schema.name in self.writers:
+            self.writers[schema.name].close()
+            del self.writers[schema.name]
+
+        output_file = open(self._get_file_name(schema.name, 'json'), 'w')  # pylint: disable=consider-using-with
```

Comment:
'w' truncates is the file exists already I think

Reply:
so that wasn't my experience before I put the explicit `truncate()` in there but will test again

Reply:
May be an issue across different platforms, Python documents it as ''w' for only writing (an existing file with the same name will be erased)'.

https://docs.python.org/3/tutorial/inputoutput.html.